src/gf.c

```diff
@@ -1518,18 +1856,17 @@ static void _compile_all_deq(jl_array_t *found)
         jl_method_t *m = ml->func.method;
         jl_method_instance_t *linfo = m->unspecialized;
         if (!linfo) {
-            // XXX: use computed env rather than empty svec
-            linfo = jl_specializations_get_linfo(m, ml->sig, jl_emptysvec);
+            linfo = jl_get_specialized(m, ml->sig, jl_emptysvec);
             m->unspecialized = linfo;
             jl_gc_wb(m, linfo);
         }
 
-        // infer this function now, if necessary
-        if (linfo->jlcall_api == 2)
-            continue;
-        src = jl_type_infer(linfo, 1);
-        if (linfo->jlcall_api == 2)
-            continue;
+        //// infer this function now, if necessary
+        //if (linfo->jlcall_api == 2)
+        //    continue;
+        //src = jl_type_infer(&linfo, jl_world_counter, 1);
+        //if (linfo->jlcall_api == 2)
+        //    continue;
```

Comment:
probably not. or at least I would need to write a new type inference mode that doesn't take `jl_world_counter` as a parameter, which could be difficult. I'll add a note to the top of the PR for future work.